### ISSUE TITLE
[BUG] Can't update config of the integration (@wills106 Potential related to 2022.12 config flow)

### ISSUE THREAD

**Issue by johny-mnemonic** [wills106/homeassistant-solax-modbus#234]
I wanted to change the polling frequency from 15 to 10 seconds, so I opened the integrations page and clicked the config button, but when I pressed send button it failed with "unknown error occurred".

**Mandatory details**
1. Integration Version: 0.6.4
2. HA Version: 2022.12.0
3. Inverter Brand: Solax X3 Gen4 10kW
4. Plugin Used:
5. First 6 digits of Serial: H34A10
6. Connection Method: ModBus TCP using WaveShare RS485 TO ETH (B)

**Detailed Error Log**
```
Logger: aiohttp.server
Source: helpers/schema_config_entry_flow.py:165
First occurred: 20:03:38 (6 occurrences)
Last logged: 20:08:00

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 81, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 136, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 241, in post
    return await super().post(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 73, in wrapper
    result = await method(view, request, data, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 281, in async_configure
    result = await self._async_handle_step(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 368, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/usr/src/homeassistant/homeassistant/helpers/schema_config_entry_flow.py", line 390, in _async_step
    result = await self._common_handler.async_step(step_id, user_input)
  File "/usr/src/homeassistant/homeassistant/helpers/schema_config_entry_flow.py", line 129, in async_step
    return await self._async_form_step(step_id, user_input)
  File "/usr/src/homeassistant/homeassistant/helpers/schema_config_entry_flow.py", line 165, in _async_form_step
    user_input = await form_step.validate_user_input(self, user_input)
TypeError: _validate_base() takes 1 positional argument but 2 were given
```

**Comment by pchaves1** [comment]
Hello,
I have the same problem. 
With the version:
![image](https://user-images.githubusercontent.com/45390982/206598681-368b27b2-e87e-4131-9519-0275447d003a.png)
I did remove the previous 7.0bx version, restarted, updated the version, restarted, and when trying to setup I get the "unknown error occurred".
Did try with several beta's and 0.6.4 and always the same problem.
Did clean the solax_modbus folder, installed manually, and same issue.
For me, the solution was to restore a backup with 12 hours (with the same version above) with a 7.0bx previously installed, but now I can't upgrade until I know what is going on.
On the log I have the same error.
I have a Solax Gen4 7.5kw and I'm also using a ModBus TCP using WaveShare RS485 TO ETH (B)

**Comment by wills106** [comment]
> change the polling frequency from 15 to 10 seconds

Normally that would give an error for reusing the same Name, but this seems to have changed on 2022.12

> TypeError: _validate_base() takes 1 positional argument but 2 were given

I didn't get this on 2022.11.5 Yesterday and I haven't upgraded to 2022.12 yet.
So I am wondering if something has changed in the handling of the config flow?

I'm assuming at the moment if you go back to 2022.11.5 and install 0.7.0b73 it will work fine?

I probably won't have time to look into this today, but if anyone can find any more info on the changes made in 2022.12 could you post links here and I'll have a look tomorrow.

**Comment by infradom** [comment]
One remark: updating the config of an integration has never been tested. Maybe we should remove this option until someone finds time to test and correct the errors. Until we find time to examine this in depth, I would recommend not to reconfigure the integration but delete it and re-create it.

**Comment by wills106** [comment]
I might be wrong, but I think there are two separate issues here.

1. Updating configuration on 2022.12 (change poll from 15 - 10)
2. Configuring a new Install on 2022.12 which still gives the "TypeError: _validate_base() takes 1 positional argument but 2 were given" error.

I'll try upgrading to 2022.12 later today / tomorrow and do a fresh install of the Integration.

**Comment by Patrik1973** [comment]
Hi,
I have similar a trouble, yesterday morning I upgraded HA to 2022.12...and than yesterday evening the integration stopped response with Solax G4. This morning I tried reinstall the integration but without success. I removed from the devices but I can't to add again.

**Comment by infradom** [comment]
I can now confirm I have the same problem after I deleted the integration and tried to re-create it in HA core version 2022.12.1

**Comment by wills106** [comment]
I have had a search on HA/Core but no similar issues yet. Might just need some time till other Integrations start reporting the same issue.

config flow hasn't changed on here recently.

**Comment by infradom** [comment]
Looking into the source code of HA, I see that there were major changes in the config_validation a couple of weeks ago.
I am trying to adapt our code now ...

**Comment by infradom** [comment]
I will try to adapt our code as soon as possible, but as I have a lunch appointment, it may take a couple of hours to test the adaptations.

**Comment by infradom** [comment]
Should be solved with pull request #236
@wills106 : can you make a new release

**Comment by wills106** [comment]
I can confirm that the changes retains support for 2022.11.5

I'll upgrade to 2022.12.x later today and confirm that works as intended.

**Comment by wills106** [comment]
At the moment on any 0.7.0bx release you need to remove / delete it from the Integration page, restart HA and reconfigure it on the Integration page.
![image](https://user-images.githubusercontent.com/18155231/206750452-5230cb76-602c-4729-b683-ef32caaac37d.png)
But for HA 2022.12 you need to be on 0.7.0b74 or later.

You don't need to actually delete the files.

**Comment by johny-mnemonic** [comment]
@wills106 that would be no problem if it would be possible to install the beta version, but the version selection in HACS is missing for this plugin :-(

**Comment by wills106** [comment]
In HACS if you goto the SolaX Integration and the 3 Dots top right. There is an option to redownload and turn on Beta Versions.

But I am almost ready to release the next stable version anyway.

**Comment by johny-mnemonic** [comment]
Well, that's the issue. It's there for other plugins but not for this one.
For others it looks like this:
![pyloxone_redownload](https://user-images.githubusercontent.com/1143074/206764902-f861c054-ad85-402c-b2ac-40e1a8007177.png)
As described [here](https://hacs.xyz/docs/faq/select_version/).
But for solax_modbus it looks like this:
![solax_modbus_redownload](https://user-images.githubusercontent.com/1143074/206764991-66d5ee48-bf79-4fa5-9a76-ae6d62314cc3.png)
🤷‍♂️

**Comment by wills106** [comment]
Have you tried "Update Information" first before "Redownload"

You should see 2022.12.1 come through shortly anyway as a stable release.

**Comment by johny-mnemonic** [comment]
It was definitely not there, so I went and installed it manually by git clone.
But now when I tried it again it is there also with the 2022.12.1. You must have fixed something 😉
Thanks a lot for a quick response!

**Comment by hallidaydaniel** [comment]
Just to confirm I tested this after experiencing the issue > updating > now fixed 👍

**Comment by Patrik1973** [comment]
When I used beta version 0.07.b74, it was until evening alright.  But when I used last version 2022.12.4, now I see 4 entities only.

**Comment by wills106** [comment]
@Patrik1973 do you have any errors in your log? Also check the extended log.

**Comment by Patrik1973** [comment]
> 

Here it is:
unrecognized inverter type - serial number : unknown
12:45:22 – (CHYBA) SolaX Inverter Modbus (vlastní integrace)
SolaX: cannot find serial number, even not for MIC
12:45:22 – (CHYBA) SolaX Inverter Modbus (vlastní integrace)
SolaX: reading serial number from address 0x300 failed; other address may succeed
12:45:22 – (VAROVÁNÍ) SolaX Inverter Modbus (vlastní integrace)
SolaX: reading serial number from address 0x0 failed; other address may succeed
12:45:22 – (VAROVÁNÍ) SolaX Inverter Modbus (vlastní integrace)

2022-12-11 12:45:22.174 WARNING (MainThread) [custom_components.solax_modbus.plugin_solax] SolaX: reading serial number from address 0x0 failed; other address may succeed
2022-12-11 12:45:22.276 WARNING (MainThread) [custom_components.solax_modbus.plugin_solax] SolaX: reading serial number from address 0x300 failed; other address may succeed
2022-12-11 12:45:22.281 ERROR (MainThread) [custom_components.solax_modbus.plugin_solax] SolaX: cannot find serial number, even not for MIC
2022-12-11 12:45:22.285 ERROR (MainThread) [custom_components.solax_modbus.plugin_solax] unrecognized inverter type - serial number : unknown
2022-12-11 12:45:37.329 ERROR (MainThread) [custom_components.solax_modbus] Modbus Error: [Connection] ModbusTcpClient(172.12.13.220:502): Connection unexpectedly closed 0.000113 seconds into unbounded read without response from unit before it closed connection: SolaX cannot read input registers at device 1 position 0x8
Traceback (most recent call last):
File "/config/custom_components/solax_modbus/__init__.py", line 362, in read_modbus_block
if typ == 'input': realtime_data = self.read_input_registers(unit=self._modbus_addr, address=block.start, count=block.end - block.start)
File "/config/custom_components/solax_modbus/__init__.py", line 287, in read_input_registers
return self._client.read_input_registers(address, count, **kwargs)
File "/usr/local/lib/python3.10/site-packages/pymodbus/client/common.py", line 125, in read_input_registers
return self.execute(request)
File "/usr/local/lib/python3.10/site-packages/pymodbus/client/sync.py", line 109, in execute
return self.transaction.execute(request)
File "/usr/local/lib/python3.10/site-packages/pymodbus/transaction.py", line 174, in execute
response, last_exception = self._transact(
File "/usr/local/lib/python3.10/site-packages/pymodbus/transaction.py", line 294, in _transact
result = self._recv(response_length, full)
File "/usr/local/lib/python3.10/site-packages/pymodbus/transaction.py", line 362, in _recv
result = self.client.framer.recvPacket(expected_response_length)
File "/usr/local/lib/python3.10/site-packages/pymodbus/framer/__init__.py", line 49, in recvPacket
return self.client.recv(size)
File "/usr/local/lib/python3.10/site-packages/pymodbus/client/sync.py", line 89, in recv
return self._recv(size)
File "/usr/local/lib/python3.10/site-packages/pymodbus/client/sync.py", line 297, in _recv
return self._handle_abrupt_socket_close(
File "/usr/local/lib/python3.10/site-packages/pymodbus/client/sync.py", line 342, in _handle_abrupt_socket_close
raise ConnectionException(msg)
pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] ModbusTcpClient(172.12.13.220:502): Connection unexpectedly closed 0.000113 seconds into unbounded read without response from unit before it closed connection
